feat: Editor

JakobVogelsang opened the pull request:
Closes #4

JakobVogelsang committed:
chore: update test and build

JakobVogelsang committed:
feat: Editor class

ca-d commented:
Sorry for taking so long to review. I have not forgotten this. @JakobVogelsang if this is still useful to you, I'll do a thorough review first thing in April. If you urgently need it sooner, please ping me again so we can discuss a time frame.

ca-d commented:
I just rebased on `main` in order to have a conflict-free request to review.

ca-d committed:
chore: fix erroneous imports

JakobVogelsang commented:
I ditched this PR to be honest with you. I do not see a need to export this class only. I think having a closure makes more sense with what we have in mind. wdyt @ca-d

ca-d committed:
test: fix attributesNS test

ca-d commented:
I have nothing against passing the class instance to plugins as is. Do you see some benefit in a closure?

stee-re commented:
Its not related to the changes made in this PR but eslint is linting everything including dist/**, so the format script must be formatting the transpiled code, which I guess is not desired.
Should be something like this:
`eslint --ignore-pattern 'dist/*.js' .`

ca-d committed:
chore: don't lint dist

ca-d committed:
chore: remove superfluous dependencies

ca-d committed:
feat(Editor): make Editor generic

Turns the previous `Editor` class which used to extend LitElement into
an `EditV2Editor` class which implements a generic `Editor` interface
and is not related to rendering at all.

ca-d commented:
We will continue this work at https://github.com/OMICRONEnergyOSS/xml-lib on the branch https://github.com/OMICRONEnergyOSS/xml-lib/tree/XMLEditor .

@JakobVogelsang We currently intend to maintain xml-lib there. How do you feel about that? Also, do you already have users of this who would be impacted by a re-naming of the npm package?

JakobVogelsang commented:
It is very ok with me. At some point, I would like us to agree on one such functionality. Where this is allocated, I do not care. I am thrilled to see our additional contribution and the many many tests you will add.
